tests/core/pyspec/eth2spec/test/helpers/fork_choice.py

```diff
@@ -144,9 +144,10 @@ def get_blobs_file_name(blobs=None, blobs_root=None):
 
 
 def on_tick_and_append_step(spec, store, time, test_steps):
-    spec.on_tick(store, time)
-    test_steps.append({'tick': int(time)})
-    output_store_checks(spec, store, test_steps)
+    if store.time < time:
```

Comment:
Shouldn’t we omit this check and let it fail? in this case it can highlight an error in a test

Reply:
This is implementation dependant, time, in particular for spectests is very tricky. Prism for example modifies genesis time on every tick check and it would be undefined behavior if it went back in time. I advocate for forkchoice tests to only fail on bad output from the API, that is, bad head, bad proposer boost roots, etc .

Reply:
Ah, I meant that if we’re adding assertion into the spec we can probably let this assertion cause an exception when the test isn’t properly design. Because if you set your time back this is probably something you’re doing wrong when writing the test, something that worth paying attention too instead of quietly omitting the problem

Reply:
If I understand correctly this is just ensuring that the tests don't get written in a way that time goes backwards, and that seems fair...

Reply:
@rolfyone is right; it was mainly to avoid righting useless steps in the existing test scripts.

About the failure cases, good point. If we want to cover it, we should add new test cases for `on_tick`. We do have the test format field to indicate if `on_tick` step is invalid: https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/fork_choice#on_tick-execution-step

But I hear you @potuz that it could be implementation-dependant...

Do you guys think it is okay to revert the spec change and *only* make the test script mistake-proofing (avoid making `time < store.time` call)? I'm trying to find the balance between the abstract FC specs and useful tests.

Reply:
An `assert` in this place looks like a nice alternative to the `assert` in the `on_tick`